src/Common/src/TypeSystem/Common/TypeSystemConstraintsHelpers.cs

```diff
@@ -142,7 +210,7 @@ public static bool CheckConstraints(this MethodDesc method)
             MethodDesc uninstantiatedMethod = method.GetMethodDefinition();
             for (int i = 0; i < uninstantiatedMethod.Instantiation.Length; i++)
             {
-                if (!VerifyGenericParamConstraint(method.OwningType.Instantiation, method.Instantiation, (GenericParameterDesc)uninstantiatedMethod.Instantiation[i], method.Instantiation[i]))
+                if (!VerifyGenericParamConstraint(new InstantiationContext(method.OwningType.Instantiation, method.Instantiation), (GenericParameterDesc)uninstantiatedMethod.Instantiation[i], context, method.Instantiation[i]))
```

Comment:
There is a new same InstantiationContext allocated for every loop iteration. It should be struct and/or allocated before the loop.